Clarify init container concept (#47946)
kubernetes/website: content/en/docs/concepts/workloads/pods/init-containers.md

If the Pod [restarts](#pod-restart-reasons), or is restarted, all init containers
must execute again.

> Changes to the init container spec are limited to the container image field. Altering the image field of the init container does not restart or recreate the Pod.

Author: It is done.

> However, since init containers can be restarted, retried, or re-executed, init container

Author: Done

> code should be idempotent. In particular, code that writes to files on `EmptyDirs`

Reviewer: (optionally)
```suggestion
code should be idempotent. In particular, code that writes into any `emptyDir` volume
```

Author: Make sense. It is done.
